chia/wallet/puzzles/genesis_checkers.py

```diff
@@ -0,0 +1,107 @@
+from typing import Optional
+from blspy import G1Element
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.wallet.puzzles.load_clvm import load_clvm
+
+GENESIS_BY_ID_MOD = load_clvm("genesis-by-coin-id-with-0.clvm")
+GENESIS_BY_PUZHASH_MOD = load_clvm("genesis-by-puzzle-hash-with-0.clvm")
+EVERYTHING_WITH_SIG_MOD = load_clvm("everything_with_signature.clvm")
+DELEGATED_GENESIS_CHECKER_MOD = load_clvm("delegated_genesis_checker.clvm")
+
+
+class GenesisById:
+    @staticmethod
+    def create(genesis_coin_id: bytes32) -> Program:
+        """
+        Given a specific genesis coin id, create a `genesis_coin_mod` that allows
+        both that coin id to issue a cc, or anyone to create a cc with amount 0.
```

Comment:
both that coin id to issue a cc of that colour, or anyone to create a cc of that colour with amount 0

Comment could be more explicit about how much of the value flows into the newly created CC

Reply:
This comment is actually no longer accurate so yeah we need to fix it up.